### PR TITLE
Do not create schema gp_toolkit within extension gp_toolkit

### PR DESCRIPTION
**Pull request by huansong**
We did that when we made gp_toolkit an extension. All objects in the extension are under 'gp_toolkit' schema, but the extension itself is created under 'public' schema (`pg_extension.extnamespace` reflects that, and there's a `pg_depend` linkage between the extension and the schema). However, conventionally 'public' schema should not contain anything other than user themselves created. And in many cases the user would actually opt to drop the 'public' schema - and they now have to drop gp_toolkit extension before doing that (despite that nothing in gp_toolkit actually belong to 'public'), which is not nice.

So now we do not create the schema within the extension, but do that via the control file which implicitly creates the schema and use it for the extension. This removes the dependency on 'public' and also conforms the conventions better. Because we cannot remove "CREATE SCHEMA gp_toolkit" from `gp_toolkit--1.0.sql`, we create a new default extension version 1.3 to achieve that.

This fixes it for new GPDB installations but existing GPDB7 installations will continue to have gp_toolkit registered under the 'public' schema which is a decision based on both impact (not a bug but rather a nice-to-have feature) and the state of 7X (not many existing installaions). If they re-create gp_toolkit extension after upgrading to a version containing this change, everything would be aligned with new installations.

~In addition, we are now dumping `gp_toolkit` schema (we used to avoid that because that's `initdb`'s work). Now we can do that because we moved that to `gpinitsystem`.~ We are dumping gp_toolkit schema during binary-upgrade, because binary-upgrade will create an empty extension with an expectation that the schema of the extension should've been there. In other cases, the gp_toolkit schema should've created with the extension itself.

For the test case, restore a regress test case that drops the 'public' schema w/o dropping anything else. 

Also, fix a separate issue in binary swap test that the database `binswap_connect` is being re-created after swapping the binary, so difference is expected. Exclude that database from `pg_dumpall`.

P.S. we continue to explicitly specify the schema name 'gp_toolkit.XXX' in the extension script, just for more clarity as many views/functions use other views/functions internally and need to have the right schema to look at things (in that regards, also removed the `relocatable ` keyword for gp_toolkit objects because they are not).

Dev pipeline: https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gp_toolkit-schema

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: gpcontrib/gp_toolkit/Makefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 EXTENSION = gp_toolkit
-DATA = gp_toolkit--1.1--1.2.sql gp_toolkit--1.0--1.1.sql gp_toolkit--1.0.sql
+DATA = gp_toolkit--1.1--1.2.sql gp_toolkit--1.0--1.1.sql gp_toolkit--1.0.sql \
```
Comment (ashwinstar) [inline gpcontrib/gp_toolkit/Makefile:2]: Should we move gp_toolkit--1.0.sql to gp_toolkit--unpackaged--1.0.sql (similar to pageinspect)?

File: src/test/regress/expected/gp_upgrade_cornercases.out [hunk 2]
```diff
@@ -52,10 +52,7 @@ DROP DATABASE IF EXISTS upgrade_recreated_public;
 NOTICE:  database "upgrade_recreated_public" does not exist, skipping
 CREATE DATABASE upgrade_recreated_public;
 \c upgrade_recreated_public
--- Some extensions are pre-baked into template1, hence the new database has
```
Comment (ashwinstar) [inline src/test/regress/expected/gp_upgrade_cornercases.out:55]: Interesting so we had faced the problem while implementing the feature?

Comment (ashwinstar) [reply src/test/regress/expected/gp_upgrade_cornercases.out:55]: what other extensions are pre-baked into template1 under public which needs this drop and create dance here? As NOTICE only conveys gp_toolkit. Like do we even need drop and create?

File: gpcontrib/gp_toolkit/gp_toolkit--1.3.sql [hunk 3]
```diff
@@ -0,0 +1,2785 @@
...
+--        oid - relation oid
+--        int - segment number
+--        eof - eof of the segment file
+--
+-- @doc:
+--        UDF to retrieve AO segment file numbers for each ao_row table
+--
+--------------------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION gp_toolkit.__get_ao_segno_list()
```
Comment (ashwinstar) [inline gpcontrib/gp_toolkit/gp_toolkit--1.3.sql:2313]: we shouldn't have `or replace` in this SQL file seems lot of instance of it which should be removed

File: src/test/regress/expected/gp_upgrade_cornercases.out (hunk 2 above)
Comment (huansong) [reply src/test/regress/expected/gp_upgrade_cornercases.out:55]: The `CASCADE` keyword was [added](https://github.com/greenplum-db/gpdb/pull/15630/files#diff-fd566dd73b595e26e7ea909f2168edd0e32494fe4ccb6bde08b8c8f1d376f242R55-R58) when moving `gp_toolkit` into extension. Though at that time I didn't view it as a problem as didn't hold the assumption about `public` schema mentioned in this PR description. No other extensions are pre-baked into template1.

> Like do we even need drop and create?

The original purpose of drop & create `public` was to prepare for some upgrade test that'll be run after regress: https://github.com/greenplum-db/gpdb/commit/5844158d125b

File: gpcontrib/gp_toolkit/Makefile (hunk 1 above)
Comment (huansong) [reply gpcontrib/gp_toolkit/Makefile:2]: `unpackaged` is the keyword for pre-9.1 postgres module when extension was not invented. The `unpackaged--1.0` script provides an upgrade path from that module into a standard extension. In our case we don't need as gp_toolkit was just recently introduced as an extension.

File: src/test/regress/expected/gp_upgrade_cornercases.out (hunk 2 above)
Comment (ashwinstar) [reply src/test/regress/expected/gp_upgrade_cornercases.out:55]: Thanks for digging the history. Please help add as comment why we wish to drop and recreate public schema here for ease in future.

File: gpcontrib/gp_toolkit/gp_toolkit--1.3.sql (hunk 3 above)
Comment (huansong) [reply gpcontrib/gp_toolkit/gp_toolkit--1.3.sql]: Agreed & done.

File: src/test/regress/expected/gp_upgrade_cornercases.out (hunk 2 above)
Comment (huansong) [reply src/test/regress/expected/gp_upgrade_cornercases.out:55]: Added a comment above to indicate purpose of the test case.

**Comment by bmdoil** [comment]
> In addition, we are now dumping gp_toolkit schema (we used to avoid that because that's initdb's work). Now we can do that because we moved that to gpinitsystem.

Curious why we're now dumping the `gp_toolkit` schema. A pg_dump of `postgres` and restore into a new database will throw an error since `gp_toolkit` exists in `template1` so database `foo` will inherit it. 

```
pg_dump --schema-only -f out.sql postgres
createdb foo
psql -f out.sql foo
psql:out.sql:24: ERROR:  schema "gp_toolkit" already exists
ALTER SCHEMA
CREATE EXTENSION
COMMENT
```

**Comment by huansong** [comment]
> Curious why we're now dumping the gp_toolkit schema. A pg_dump of postgres and restore into a new database will throw an error since gp_toolkit exists in template1 so database foo will inherit it.

Yeah that's problematic. The reason for that was to avoid an issue in `pg_upgrade` test: when we are binary-upgrading an extension, we create an empty extension first which only inserts a `pg_extension` row(see `binary_upgrade_create_empty_extension()`), so we expect the schema to have been there (we need to fill the `pg_extension.extnamespace` column for the extension). It seems I was over-fixing it: only during binary upgrade we expect the schema to be there. Otherwise, the schema should've created with the extension itself (no other cases creates the empty extension). I've updated the fix.

**Pull request merged**